perllib/FixMyStreet/App/Controller/Auth.pm

```diff
@@ -239,6 +239,7 @@ sub email_sign_in : Private {
 
     $c->stash->{token} = $token_obj->token;
     my $template = $c->stash->{email_template} || 'login.txt';
+    $c->cobrand->call_hook('disable_login_for_email', $good_email) unless $c->get_param('oauth_need_email');
```

Comment:
I think this should be higher, why generate the token etc if going to bail out with this line? Perhaps as soon as good_email is fetched? Unless there's a reason it's here.